api.go

```diff
@@ -35,27 +40,36 @@ func Home(w http.ResponseWriter, r *http.Request) {
 	return
 }
 
-// RenderJSON helder for rndering JSON response, it marshals value into json and writes
+// RenderJSON helper for rendering JSON response, it marshals value into json and writes
 // it into w.
 func RenderJSON(w http.ResponseWriter, code int, value interface{}) {
 	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(code)
 	err := json.NewEncoder(w).Encode(value)
 	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+		http.Error(w, err.Error(), code)
```

Comment:
This error happened when encoding the value internaly. I think  `StatusInternalServerError` is the right status for this since it has nothing to do with the api consumer.